src/components/additional-buttons-homepage.tsx

```diff
@@ -24,8 +31,10 @@ const Buttons = ({ session }: { session: Session | null }) => {
               <ArrowRightDarkModeFriendly />
             </Button>
             <Button
-              onClick={() =>
-                signIn("azure-ad", { callbackUrl: "/find-the-expert/general" })
+              onClick={async () => {
+                  handleLogging();
+                  await signIn("azure-ad", { callbackUrl: "/find-the-expert/general" });
```

Comment:
I'll leave this as a suggestion, but what you can also do here is extract everything to a seperate handle, so also the signIn that was there earlier. Separating the logic from the HTML typically makes the React code cleaner and easier to maintain/read.